### PR TITLE
Added Notification Engine

### PR DESCRIPTION
**Pull request by rajatgl17**

### CODE REVIEW

File: plato-adapter/src/main/java/org/techpleiad/plato/adapter/config/EmailConfig.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Configuration;
+import org.techpleiad.plato.core.port.out.IEmailConfigurationPort;
+
+@RefreshScope
+@Configuration
+@EnableConfigurationProperties
+@Data
+public class EmailConfig implements IEmailConfigurationPort {
```
Comment (rajatgl17) [inline plato-adapter/src/main/java/org/techpleiad/plato/adapter/config/EmailConfig.java:16]: This class is not required

File: plato-adapter/src/main/java/org/techpleiad/plato/adapter/web/in/ServiceManagerController.java [hunk 2]
```diff
@@ -91,6 +98,12 @@ public ResponseEntity addService(@Valid final ServiceRequestTO serviceRequestTO)
                         servicesAcrossProfileValidateRequestTO.isIncludeSuppressed()
                 );
 
+        String mailBody = htmlServiceUseCase.createProfileReportMailBody(reportList, branchName);
```
Comment (rajatgl17) [inline plato-adapter/src/main/java/org/techpleiad/plato/adapter/web/in/ServiceManagerController.java:101]: inside if statement

File: plato-adapter/src/main/java/org/techpleiad/plato/adapter/web/in/ServiceManagerController.java [hunk 3]
```diff
@@ -119,6 +132,13 @@ public ResponseEntity validateAcrossBranches(@Valid final ServicesAcrossBranchVa
                 validationAcrossBranchConfig
         );
 
+        String mailBody = htmlServiceUseCase.createBranchReportMailBody(reportList,acrossBranchValidateRequestTO.getFromBranch(),
```
Comment (rajatgl17) [inline plato-adapter/src/main/java/org/techpleiad/plato/adapter/web/in/ServiceManagerController.java:135]: inside if statement

File: plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.List;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Getter
+@ToString
+public class EmailConfigRequestTo {
+    private boolean sendEmail;
+    private List<String> recipients;
```
Comment (rajatgl17) [inline plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java:19]: Use annotation @ValidEmail

File: plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.ToString;
+
+import java.util.List;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Getter
+@ToString
+public class EmailConfigRequestTo {
```
Comment (rajatgl17) [inline plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java:17]: TO

Comment (rajatgl17) [reply plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java:17]: EmailRequestTO

File: plato-core/src/main/java/org/techpleiad/plato/core/exceptions/EmailException.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import lombok.Getter;
+import lombok.ToString;
+
+@ToString
+@Getter
+public class EmailException extends RuntimeException {
+    private String errorMessage;
+
+    public EmailException(final String errorMessage) {
```
Comment (rajatgl17) [inline plato-core/src/main/java/org/techpleiad/plato/core/exceptions/EmailException.java:11]: Not required

File: plato-core/src/main/java/org/techpleiad/plato/core/port/out/IEmailConfigurationPort.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package org.techpleiad.plato.core.port.out;
+
+public interface IEmailConfigurationPort {
```
Comment (rajatgl17) [inline plato-core/src/main/java/org/techpleiad/plato/core/port/out/IEmailConfigurationPort.java:3]: Not required

File: plato-core/src/main/java/org/techpleiad/plato/core/service/EmailService.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+import java.util.Properties;
+
+@Service
+@Slf4j
+public class EmailService implements IEmailServiceUseCase {
+
+    @Autowired
+    private IEmailConfigurationPort emailConfiguration;
+
```
Comment (rajatgl17) [inline plato-core/src/main/java/org/techpleiad/plato/core/service/EmailService.java:26]: Use spring email

File: plato-core/src/main/java/org/techpleiad/plato/core/service/HtmlService.java [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    private static final String LEGENDPROFILE = "<br><table style=\" border: 1px solid black; border-collapse: collapse;\"><tr><th>Legend</th></tr><tr><td style=\"background-color: red;  border: 1px solid black; border-collapse: collapse; padding: 15px; text-align: left;\"></td><td style=\"border: 1px solid black; border-collapse: collapse; padding: 15px; text-align: left;\">Inconsistent with other profiles</td></tr><tr><td style=\"background-color: green; border: 1px solid black; border-collapse: collapse; padding: 15px; text-align: left;\"></td><td style=\"border: 1px solid black; border-collapse: collapse; padding: 15px; text-align: left;\">Consistent and good to go</td></tr></table>";
+
+    public String createBranchReportMailBody(final List<ConsistencyAcrossBranchesReport> reportList, final String fromBranch, final String toBranch) {
+        String htmlDocument = "<h1>Branch Consistency Report<h1>";
```
Comment (rajatgl17) [inline plato-core/src/main/java/org/techpleiad/plato/core/service/HtmlService.java:26]: @Override and private

File: plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java (hunk 4 above)
Comment (DialgicMew) [reply plato-api/src/main/java/org/techpleiad/plato/api/request/EmailConfigRequestTo.java:19]: Will have to add new library or write custom code for validation of email.

**Pull request merged**